android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt

```diff
@@ -276,8 +276,22 @@ class RNOneginiSdk(reactContext: ReactApplicationContext) : ReactContextBaseJava
 
     @ReactMethod
     fun cancelRegistration(promise: Promise) {
-        oneginiSDK.registrationRequestHandler.cancelRegistration()
-        promise.resolve(null)
+        var canceled = oneginiSDK.registrationRequestHandler.cancelRegistration()
```

Comment:
Create one method called `shouldRegistrationBeCanceled()` and put all of those conditions in there, so here you're gonna have 

```
var shouldBeCanceled = shouldRegistrationBeCanceled()
if (shouldBeCanceled) {
...
}
```